Doc/c-api/unicode.rst

```diff
@@ -1538,7 +1538,8 @@ object.
 
    Write the single Unicode character *ch* into *writer*.
 
-   Return ``0`` on success, or set an exception and return ``-1`` on error.
+   On success, return ``0``.
+   On error, set an exception, leave the writer unchanged, and return ``-1``.
```

Comment:
It is impossible to guarantee this with the current code.

Writing even a single character can require resizing the Unicode object, which can fail, and you will lose the whole buffer.

You can only make it atomic if you always create a new Unicode object instead of resizing in-place, but this is less efficient. I do not think that we should sacrifice performance for this.

Reply:
You can ignore this comment. It seems I was wrong.